pkg/controller/operators/olm/operator.go

```diff
@@ -1996,6 +1996,16 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		}
 
 	case v1alpha1.CSVPhaseFailed:
+		// TODO(fail-forward): If the csv allows fail forward
+		// Transition to replacing if FailForward is enabled, a CSV exists that replaces the operator, and the operator being replaced supports failing forward.
+		if operatorGroup.UpgradeStrategy() == operatorsv1.UnsafeFailForwardUpgradeStrategy {
+			if replacement := a.isBeingReplaced(out, a.csvSet(out.GetNamespace(), v1alpha1.CSVPhaseAny)); replacement != nil {
+				msg := fmt.Sprintf("%s csv being replaced by csv: %s", out.Status.Phase, replacement.GetName())
+				out.SetPhaseWithEvent(v1alpha1.CSVPhaseReplacing, v1alpha1.CSVReasonBeingReplaced, msg, a.now(), a.recorder)
```

Comment:
Given we set an event here, I would like to see the msg to reflect this is unsafeupgrade mode, not just standard transition phase. WDYT?

Reply:
That sounds reasonable to me!